### PR TITLE
GameEnd Tweaks

### LINKED ISSUE #463: Generator issue on win
**Describe the bug**
When you win a game and enter the generator, you collect the materials left over from the game and they have custom names such as custom0, custom1, etc.

Youtube video: https://youtu.be/1lMN5DCsBbE

**To Reproduce**
1. Join a game
2. Win the game
3. Uncollected items remain in the generator
4. When you pickup them they have custom names like custom0, etc

**Expected behavior**

**Screenshots**

**System (please complete the following information):**
 - OS: Linux
 - Server software: Paper
 - Server Version: 1.8.8
 - Plugin Version: Bedwars1058 22.3.4
**Additional context**

### PR DESCRIPTION
**Pull request by JT122406**
Stops Generator from dropping items and prevent winning alive players from picking up generator items and clear inventory
Fixes #463

### CODE REVIEW

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.util.UUID;
+
+public class GameEnd implements Listener {
+
+    @EventHandler
+    public void stopPickUP(GameEndEvent event){
+        if (!event.getAliveWinners().isEmpty()){
+            for (UUID p : event.getAliveWinners()) {
+                Bukkit.getPlayer(p).setCanPickupItems(false);
```
Comment (andrei1058) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java]: I'm unsure that would work on all mc versions

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                Bukkit.getPlayer(p).getInventory().clear();
+            }
+                for (IGenerator gen : event.getArena().getOreGenerators()) {
```
Comment (andrei1058) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java]: we can simply do a check on arena state in generator#spawn

**Review by andrei1058 — CHANGES_REQUESTED** [review]

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java (hunk 1 above)
Comment (JT122406) [reply bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java]: I had people test it and it seems to work fine on all versions at least I know it works on 1.8.8 and 1.17

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java (hunk 2 above)
Comment (JT122406) [reply bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java]: I have addressed this

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java (hunk 1 above)
Comment (JT122406) [reply bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java]: I have addressed this issue

**Comment by andrei1058** [comment]
there's no need to register 100 pick up listeners. this must be implemented in common module.
https://github.com/andrei1058/BedWars1058/blob/e11450381d635ccbbf771b315b5b88a2c0540052/versionsupport_common/src/main/java/com/andrei1058/bedwars/listeners/ItemDropPickListener.java#L134

**Pull request merged**